src/app/clusters/icd-management-server/icd-management-server.cpp

```diff
@@ -137,76 +149,213 @@ CHIP_ERROR IcdManagementAttributeAccess::ReadRegisteredClients(EndpointId endpoi
+    }
+    else if (CHIP_ERROR_NOT_FOUND == err)
+    {
+        // New entry
+        VerifyOrReturnError(entry.index < table.Limit(), InteractionModel::Status::ResourceExhausted);
+
+        // Check if it's going to be the first entry for fabric
+        isFirstEntryForFabric = table.IsEmpty();
+    }
+    else
+    {
+        // Error
+        return InteractionModel::Status::Failure;
+    }
+
+    // Save
+    entry.checkInNodeID    = node_id;
+    entry.monitoredSubject = monitored_subject;
+    if (entry.keyHandleValid)
+    {
+        entry.DeleteKey();
+    }
+
+    err = entry.SetKey(key);
+    VerifyOrReturnError(CHIP_ERROR_INVALID_ARGUMENT != err, InteractionModel::Status::ConstraintError);
+    VerifyOrReturnError(CHIP_NO_ERROR == err, InteractionModel::Status::Failure);
+    err = table.Set(entry.index, entry);
+
+    // Delete key upon failure to prevent key storage leakage.
+    if (err != CHIP_NO_ERROR)
+    {
+        entry.DeleteKey();
+    }
+
+    VerifyOrReturnError(CHIP_ERROR_INVALID_ARGUMENT != err, InteractionModel::Status::ConstraintError);
+    VerifyOrReturnError(CHIP_NO_ERROR == err, InteractionModel::Status::Failure);
+
+    if (isFirstEntryForFabric)
+    {
+        // Notify subscribers that the first entry for the fabric was successfully added
+        TriggerICDMTableUpdatedEvent();
+    }
+
+    icdCounter = mICDData->GetICDCounter();
+    return InteractionModel::Status::Success;
 }
 
+Status ICDManagementServer::UnregisterClient(FabricIndex fabric_index, NodeId node_id, Optional<ByteSpan> verificationKey,
+                                             bool isAdmin)
+{
+    ICDMonitoringTable table(*mStorage, fabric_index, mICDData->GetClientsSupportedPerFabric(), mSymmetricKeystore);
+
+    // Get current entry, if exists
+    ICDMonitoringEntry entry(mSymmetricKeystore);
+    CHIP_ERROR err = table.Find(node_id, entry);
+    VerifyOrReturnError(CHIP_ERROR_NOT_FOUND != err, InteractionModel::Status::NotFound);
+    VerifyOrReturnError(CHIP_NO_ERROR == err, InteractionModel::Status::Failure);
+
+    // Existing entry: Validate Key if, and only if, the ISD has NOT administrator permissions
+    if (!isAdmin)
+    {
+        VerifyOrReturnError(verificationKey.HasValue(), InteractionModel::Status::Failure);
+        VerifyOrReturnError(entry.IsKeyEquivalent(verificationKey.Value()), InteractionModel::Status::Failure);
+    }
+
+    err = table.Remove(entry.index);
+    VerifyOrReturnError(CHIP_NO_ERROR == err, InteractionModel::Status::Failure);
+
+    if (table.IsEmpty())
+    {
+        TriggerICDMTableUpdatedEvent();
+    }
+
+    return InteractionModel::Status::Success;
+}
+
+Status ICDManagementServer::StayActiveRequest(FabricIndex fabric_index)
+{
+    // TODO: Implementent stay awake logic for end device
+    // https://github.com/project-chip/connectedhomeip/issues/24259
+    app::ICDNotifier::GetInstance().BroadcastICDManagementEvent(app::ICDListener::ICDManagementEvents::kStayActiveRequestReceived);
```

Comment:
I don't think you need app:: here and other places. You have `using namespace chip::app;` in this file already
```suggestion
  ICDNotifier::GetInstance().BroadcastICDManagementEvent(ICDListener::ICDManagementEvents::kStayActiveRequestReceived);
```

Reply:
done